### PR TITLE
feat(complex voucher)currency changes affect transaction's amounts

### LINKED ISSUE #2953: Converting monetary values via Journal Voucher
This issue proposes to be able to convert the already pre-filled data into the Journal Grid voucher whenever you change currency,

There is a difficulty for some toolkit such as the payment of agreements, the payment of wages of employees that the company or the conventions makes their payment with a currency other than the currency of the company, and if one change the currency the monetary value already pre-filled in the grid does not change

### PR DESCRIPTION
**Pull request by jeremielodi**
closes #2953

### CODE REVIEW

**Comment by lomamech** [comment]
:boom:

File: client/src/modules/vouchers/complex-voucher.ctrl.js [hunk 1]
```diff
@@ -31,6 +32,10 @@ function ComplexJournalVoucherController(
   vm.itemIncrement = 1;
   vm.timestamp = new Date();
 
+  vm.currentCurrency = {
```
Comment (mbayopanda) [inline client/src/modules/vouchers/complex-voucher.ctrl.js:35]: 👍

File: client/src/modules/vouchers/complex-voucher.ctrl.js [hunk 2]
```diff
@@ -46,6 +51,28 @@ function ComplexJournalVoucherController(
     vm.gridApi.core.notifyDataChange(uiGridConstants.dataChange.EDIT);
   };
 
+  vm.currencyChange = function currencyChange(vCurrencyId) {
```
Comment (mbayopanda) [inline client/src/modules/vouchers/complex-voucher.ctrl.js:54]: 👍 very well

**Review by mbayopanda — COMMENTED** [review]
Well done 👍

**Comment by jniles** [comment]
I'm confused - where do we consider the voucher date?   If we set the date to a year ago, does it use today's exchange rate instead of the rate a year ago?

**Pull request merged**